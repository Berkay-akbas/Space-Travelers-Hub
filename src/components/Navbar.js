import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../assests/logo.png';
import '../style/navbar.css';

const Navbar = () => (
  <div className="homeNav">
    <div className="navlinks">
      <img className="logo" src={logo} alt="logo" />
      <h1 className="title">Space Travelers&apos; Hub</h1>
      <ul className="holder">
        <li>
          <NavLink to="/" className="link">Rockets</NavLink>
        </li>
        <li>
          <NavLink to="/missions" className="link">Missions</NavLink>
        </li>
        <li>
          |
        </li>
        <li>
          <NavLink to="/my-profile" className="link">My Profile</NavLink>
        </li>
      </ul>
    </div>
    <hr />
  </div>
);

export default Navbar;
