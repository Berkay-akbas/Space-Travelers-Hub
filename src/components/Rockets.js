import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchData } from '../redux/rockets/rocket';
import RocketItem from './RocketItem';
import '../style/rocket.css';

const Rockets = () => {
  const lists = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchData());
  }, []);

  return (
    <>
      <div className="rocketContiner">
        {
          lists.map((item) => (<RocketItem rockets={item} key={item.id} />))
        }
      </div>
    </>
  );
};

export default Rockets;
