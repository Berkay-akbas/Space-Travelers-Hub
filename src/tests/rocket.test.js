import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter as Router } from 'react-router-dom';

import { Provider } from 'react-redux';
import Navbar from '../components/Navbar';
import Rockets from '../components/Rockets';

import store from '../redux/configureStore';

it('Navbar component test', () => {
  const navbarTest = renderer.create(<Router><Navbar /></Router>).toJSON();
  expect(navbarTest).toMatchSnapshot();
});

it('Rocket component test', () => {
  const rocketTest = renderer.create(<Provider store={store}><Router><Rockets /></Router></Provider>).toJSON();
  expect(rocketTest).toMatchSnapshot();
});
