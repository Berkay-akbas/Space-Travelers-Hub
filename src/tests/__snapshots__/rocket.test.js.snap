// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component test 1`] = `
<div
  className="homeNav"
>
  <div
    className="navlinks"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <h1
      className="title"
    >
      Space Travelers' Hub
    </h1>
    <ul
      className="holder"
    >
      <li>
        <a
          aria-current="page"
          className="link active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        |
      </li>
      <li>
        <a
          className="link"
          href="/my-profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
  <hr />
</div>
`;

exports[`Rocket component test 1`] = `
<div
  className="rocketContiner"
/>
`;
